packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -193,104 +202,106 @@ function translateWithDictionary(
   }
 
   const processedString =
-    typeof result === 'string' && pseudotranslate
+    isString(result) && pseudotranslate
       ? pseudotranslateString(result, {
           ...PSEUDOTRANSLATE_OPTIONS,
           toLocale:
             typeof pseudotranslate === 'boolean' ? undefined : pseudotranslate,
         })
       : result;
 
-  if (typeof processedString === 'string') {
-    return updateStringWithReplacements(processedString, {
-      ...replacements,
-      ...additionalReplacements,
-    });
-  } else {
+  if (!isString(processedString)) {
     return MISSING_TRANSLATION;
   }
+
+  return updateStringWithReplacements(
+    processedString,
+    {
+      ...replacements,
+      ...additionalReplacements,
+    },
+    {interpolate},
+  );
 }
 
+type UpdateStringWithReplacementsOptions = Pick<
+  TranslateOptions,
+  'interpolate'
+>;
+
 function updateStringWithReplacements(
   str: string,
   replacements?: ComplexReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): React.ReactElement<any>;
 function updateStringWithReplacements(
   str: string,
   replacements?: PrimitiveReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): string;
 function updateStringWithReplacements(
   str: string,
   replacements:
     | ComplexReplacementDictionary
     | PrimitiveReplacementDictionary = {},
+  {interpolate}: UpdateStringWithReplacementsOptions = {},
 ): any {
-  const replaceFinder = /([^{]*)({([^}]*)})?/g;
-  const allReplacementsArePrimitives = Object.keys(replacements).every(
-    (key) => typeof replacements[key] !== 'object',
-  );
-
-  if (allReplacementsArePrimitives) {
-    return str.replace(REPLACE_REGEX, (match) => {
-      const replacement = match.substring(1, match.length - 1);
```

Comment:
This line assumes the `RegExp` value and the placeholder format. With my suggested changes, the `RegExp` is now the only source of truth.